File: 算法/02_remove_elements.py
class Solution:
    def removeElement(self, nums: list[int], val: int) -> int:
        slowIndex = 0
        fastIndex = 0
        for i in nums:
            if(nums[fastIndex] != val):
                nums[slowIndex] = nums[fastIndex]
                slowIndex+=1
            fastIndex+=1
                
        return slowIndex

File: 算法/02_remove_elements.cpp
#include<iostream>
#include<vector>
using namespace std;
class Solution
{
    public:

        int removeElement(vector<int>& nums ,int val)
        {
            int slowIndex = 0;
            for(int fastIndex = 0;fastIndex<nums.size();fastIndex++)
            {
                if(nums[fastIndex] != val)
                {
                    nums[slowIndex++] = nums[fastIndex];
                } 
            }     
            return slowIndex;   
        } 

};
int main()
{
    vector<int> test;
    test.push_back(3);
    test.push_back(2);
    test.push_back(2);
    test.push_back(3);

    Solution sol;
    sol.removeElement(test,2);
    // cout<<"hello"<<endl;

}